### PR TITLE
NAS-103387 / 12 / Add authorized keys to config

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/config.py [hunk 1]
```diff
@@ -16,7 +16,9 @@
 from middlewared.plugins.pwenc import PWENC_FILE_SECRET
 from middlewared.plugins.pool import GELI_KEYPATH
 
-CONFIG_FILES = {'pwenc_secret': PWENC_FILE_SECRET, 'geli': GELI_KEYPATH}
+CONFIG_FILES = {
+    'pwenc_secret': PWENC_FILE_SECRET, 'geli': GELI_KEYPATH, 'authorized_keys': '/root/.ssh/authorized_keys'
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/config.py]: Can this be formatted properly please?

File: src/middlewared/middlewared/plugins/config.py [hunk 2]
```diff
@@ -28,6 +32,7 @@ class ConfigService(Service):
         'configsave',
         Bool('secretseed', default=False),
         Bool('pool_keys', default=False),
+        Bool('authorized_keys', default=False),
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/config.py]: Maybe root_authorized_keys to make it clearer?

Also need the docstring updated.

File: src/middlewared/middlewared/plugins/config.py [hunk 3]
```diff
@@ -37,6 +42,8 @@ class ConfigService(Service):
         `secretseed` will include the password secret seed in the bundle.
 
         `pool_keys` when set will include the geli encryption keys in the bundle.
+
+        `root_authorized_keys` when set will include "root" users "authorized_keys" file in the debug.
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/config.py]: This is not a debug

File: src/middlewared/middlewared/plugins/config.py [hunk 4]
```diff
@@ -37,6 +42,8 @@ class ConfigService(Service):
         `secretseed` will include the password secret seed in the bundle.
 
         `pool_keys` when set will include the geli encryption keys in the bundle.
+
+        `root_authorized_keys` when set will include "root" users "authorized_keys" file in the bundle.
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/config.py]: Just noticed `"root" users` also sounds strange. Perhaps ask docs review :P

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/config.py]: :P

Comment (wblock) [reply src/middlewared/middlewared/plugins/config.py]: This is really unclear.  Is it supposed to be possessive?  In that case, avoid possessives, they are often unclear.  "When set" is not needed in any of these, and using present tense (include) is clearer than future tense (will include).  Can also factor out "in the bundle", like
```
Create a bundle of security-sensitive information. These options select which information is included in the bundle:

`secretseed`: include password secret seed.
`pool_keys`: include GELI encryption keys.
`root_authorized_keys`: include "authorized_keys" file for the root user.
```

**Comment by sonicaj** [comment]
@wblock so if no of the options is selected, we wouldn't have a bundle. Just the database file. Can you please advise how we should factor that in the documentation ? Thank you

**Comment by william-gr** [comment]
@wblock ping

**Pull request merged**